### PR TITLE
Share Collection and Share Dataset features

### LINKED ISSUE #553: Collection page: Share
The goal is to replicate the Share feature on the Collection Page, supporting the same sharing channel options as in JSF.

### PR DESCRIPTION
**Pull request by g-saracca**
## What this PR does / why we need it:
Replicates the Share feature on the Collection Page and Dataset Page, supporting the same sharing channel options as in JSF.
Same modal component `SocialShareModal` it's used for both pages.

## Which issue(s) this PR closes:

- Closes #553 

## Suggestions on how to test this:
Click the share buttons and check that the redirections are correct.
Unlike the JSF version, the links do not open in a popup but open in a new tab.

## Does this PR introduce a user interface change? If mockups are available, please link/include them here:
Yes.
Collection Page
Screen Shot 2024-11-21 at 14 43 15

Dataset Page
Screen Shot 2024-11-21 at 14 43 28
Screen Shot 2024-11-21 at 14 43 36

## Is there a release notes update needed for this change?:
No

## Additional documentation:
No

### CODE REVIEW

**Comment by g-saracca** [comment]
@pdurbin as promised, here is the [link](https://646f68aa9beb01b35c599acd-ptyfmzelse.chromatic.com/?path=/story/sections-shared-socialsharemodal--share-collection) to the `SocialShareModal` component storybook.

**Comment by pdurbin** [comment]
@g-saracca looks good to me! Modern icons!

File: tests/component/sections/shared/social-share-modal/SocialShareModal.spec.tsx [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+        title="The Title"
+        helpText="The Help Text"
+        shareUrl="shareUrl"
+        handleClose={() => {}}
+      />
+    )
+
+    cy.findByText('The Title').should('exist')
+    cy.findByText('The Help Text').should('exist')
+    cy.get('a').should('have.length', 3)
```
Comment (ChengShi-1) [inline tests/component/sections/shared/social-share-modal/SocialShareModal.spec.tsx:17]: we could also check if each social link has correct href, and check when` show = {false}`

**Review by ChengShi-1 — CHANGES_REQUESTED** [review]
Great job~ I left some comments here may need changes.

File: tests/component/sections/shared/social-share-modal/SocialShareModal.spec.tsx (hunk 1 above)
Comment (g-saracca) [reply tests/component/sections/shared/social-share-modal/SocialShareModal.spec.tsx:17]: Thanks, will add links assertion, the close and open functionality (`show` boolean) is tested in the collection page spec

**Comment by g-saracca** [comment]
@ChengShi-1 all the requested changes have been addressed 👍🏼 , thanks!

**Review by ChengShi-1 — DISMISSED** [review]
Good! Approve

**Comment by g-saracca** [comment]
Hi @ofahimIQSS let me update this branch with latest from develop so you can try again 👍🏼

**Comment by g-saracca** [comment]
@ofahimIQSS branch updated, its a weird error and not related to the changes applied in this feature but try again now and let me know, thanks!

**Review by ChengShi-1 — APPROVED** [review]

**Comment by ofahimIQSS** [comment]
Verified in local - no issues found. Merging PR 
image
image

**Pull request merged**